### PR TITLE
Added privKeyToHdWallet() function.

### PR DESCRIPTION
**Pull request by dplusplus1024**
Added a function to return an HD wallet, given 256 bits of entropy and a passphrase.

### CODE REVIEW

File: code/BTCUtil.c [hunk 1]
```diff
@@ -780,6 +780,17 @@ Datas ckdIndicesFromPath(const char *cStr)
     return DTPopDatas(result);
 }
 
+Data privKeyToHdWallet(Data privKey, const char *passphrase)
+{
```
Comment (ddustin) [inline code/BTCUtil.c:784]: Add a `DataTrackPush();` on the first line. This "indents" the ARC scope by one -- it's not strictly needed but doing it allows the functions in this file (that require being inside an ARC scope) to be called by code not in any ARC scope.

File: code/BTCUtil.c [hunk 2]
```diff
@@ -780,6 +780,17 @@ Datas ckdIndicesFromPath(const char *cStr)
...
+    Data seedPhrase = DataCopy("Bitcoin seed", strlen("Bitcoin seed"));
+
+    Data seed = PBKDF2(toMnemonic(privKey).bytes, passphrase);
+
+    Data hash = hmacSha512(seedPhrase, seed);
+
+    return hdWalletPriv(DataCopyDataPart(hash, 0, 32), DataCopyDataPart(hash, 32, 32));
```
Comment (ddustin) [inline code/BTCUtil.c]: To "unindent" the ARC scope by one, this needs to become `return DTPop(...);` where ... is your hdWalletPriv code.

File: code/BTCUtil.c (hunk 1 above)
Comment (ddustin) [inline code/BTCUtil.c:784]: Let's add a check here for `if(toMnemonic(privKey).bytes == NULL)` and if it is null return `DTPopNull();`

File: code/BTCUtil.c [hunk 3]
```diff
@@ -780,6 +780,17 @@ Datas ckdIndicesFromPath(const char *cStr)
...
 }
 
+Data privKeyToHdWallet(Data privKey, const char *passphrase)
+{
+    Data seedPhrase = DataCopy("Bitcoin seed", strlen("Bitcoin seed"));
+
+    Data seed = PBKDF2(toMnemonic(privKey).bytes, passphrase);
+
+    Data hash = hmacSha512(seedPhrase, seed);
+
```
Comment (ddustin) [inline code/BTCUtil.c:800]: Before we pull specific indices off `hash`, we should confirm it is at least that big (lest we create a buffer overrun). Add an assert like so `BTCUTILAssert(hash.length == 64);`

Comment (ddustin) [inline code/BTCUtil.c:800]: After the assert, add another line for safety when asserts are disabled:
`if(hash.length != 64)
return DTPopNull();`

**Review by ddustin — CHANGES_REQUESTED** [review]
Love the name. Needs safety code added. You can see examples of the safety code in other functions in this file.

**Comment by dplusplus1024** [comment]
Okay, it's been updated with all the safety precautions.

File: code/BTCUtil.c [hunk 4]
```diff
@@ -780,6 +780,29 @@ Datas ckdIndicesFromPath(const char *cStr)
...
+    Data seed = PBKDF2(toMnemonic(privKey).bytes, passphrase);
+
+    Data hash = hmacSha512(seedPhrase, seed);
+    
+    BTCUTILAssert(hash.length == 64);
+
+    if(hash.length != 64)
+        return DTPopNull();
+
+    Data hdWallet = hdWalletPriv(DataCopyDataPart(hash, 0, 32), DataCopyDataPart(hash, 32, 32));
```
Comment (ddustin) [inline code/BTCUtil.c]: This variable name needs to be named something difference since `hdWallet(...)` is already a function name

**Review by ddustin — CHANGES_REQUESTED** [review]
Love it. One last change and it should be perfect 👍

**Review by ddustin — APPROVED** [review]

**Pull request merged**